### ISSUE TITLE
Add RandomSnow preprocessing layer

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#357]
## Weather Augmentation

One of the real-world scenarios that pose challenges for training neural networks of Autonomous vehicles

![output_20_0](https://user-images.githubusercontent.com/17668390/164973749-686a78cc-378f-4e76-b99f-cb488b76ae9a.png)

Ref. Imple.

- https://albumentations.ai/docs/api_reference/augmentations/transforms/#albumentations.augmentations.transforms.RandomSnow
- https://github.com/UjjwalSaxena/Automold--Road-Augmentation-Library

**Comment by bhack** [comment]
@LukeWood  Do we have any experimental evidence that KLP layers like this one are really going to improve the performance on real snow images? 
In many of these KLP layers we don't have any academic reference and also it seems to me that they are going to create enough artifacts.
I suppose that we don't want to just collect as many augmentation we can but we want to give to the users a "curated" selection of augmentations with enough confirmed experimental results.

**Comment by bhack** [comment]
I say this because I think that giving a false expectation to users about our confidence in the results of an augmentation operation is counterproductive for the library.

**Comment by bhack** [comment]
I think also that my comments above are quite aligned with my interpretation of the [Keras-cv Roadmap](https://github.com/keras-team/keras-cv/blob/master/.github/ROADMAP.md#non-goals):


>KerasCV is not a research library.
Researchers may use it, but we do not consider researchers to be our target audience. Our target audience is applied CV engineers with experimentation and production needs. KerasCV should make it possible to quickly reimplement industry-strength versions of the latest generation of architectures produced by researchers, but we don't expect the research effort itself to be built on top of KerasCV. This enables us to focus on usability and API standardization, and produce objects that have a longer lifespan than the average research project.

**Comment by innat** [comment]
@bhack agrees for the most part. The weather-type image processing is very specialized. We may not think of it until heavily need them, i.e [weather classification](https://arxiv.org/pdf/1808.00588.pdf). 

I hardly found strong academic work on them to provide experimental evidence. I did a test once on a [wheat detection](https://www.kaggle.com/code/ipythonx/keras-global-wheat-detection-with-mask-rcnn) problem, where applying weather augmentation (rani/snow), did improve the local CV.

**Comment by bhack** [comment]
We have https://acdc.vision.ee.ethz.ch/ that is also part of CVPR [ACDC Challenge 2022](https://acdc.vision.ee.ethz.ch/news#challenge2022). Weather variations is subset of the advert challenges.

In the RGBD regime we had some results in 2019 for rain (30+ citations) https://www.embedded.uni-tuebingen.de/assets/publications/Volk-Mueller-vonBernuth-Bringmann_CNN_Augmentation.pdf.

**Comment by bhack** [comment]
See also CVPR 2022 https://vision4allseason.net/

**Comment by bhack** [comment]
P.s. here we could monitor the ACDC dataset paper citations: 
https://scholar.google.com/scholar?cites=8447842504056485417

**Comment by LukeWood** [comment]
If we strong references regarding the usefulness of these lets re-open, but for now I'd like to wait for a user feature request
